### DISCUSSION TITLE
How do I set up `pre-commit` for dev in a conda environment?

### DISCUSSION THREAD

**Discussion by chloeh13q** [ibis-project/ibis#6689] [Q&A] [answered]
I had trouble setting up the development environment via `nix` (not sure if it's because I have a MacBook with an M1 chip), so I set up everything in a conda environment instead. However, I had problems with a few pre-commit hooks, most notably `ruff`, with the following error:
```
ruff.....................................................................Failed
- hook id: ruff
- exit code: 2

error: TOML parse error at line 389, column 3
    |
389 |   "FA",  # flake8-future-annotations
    |   ^^^^
Unknown rule selector `A`
```

@deepyaman gets something similar in his dev environment:
```
ruff.....................................................................Failed
- hook id: ruff
- exit code: 2

error: TOML parse error at line 431, column 3
    |
431 |   "PLC0105", # covariant type parameters should have a _co suffix
    |   ^^^^^^^^^
Unknown rule selector: `PLC0105`
```

Has anyone successfully set up all the pre-commit hooks for Ibis in a conda environment?

**Comment by jcrist** [comment, marked as answer]
You'll see those kind of errors when your version of `ruff` is too old. Two things that could cause this:

- You may just need to update your conda environment with recently generated lockfiles.
- If you're using the latest conda-lock files then the issue is that those can be a bit out of date. The conda-lock files aren't always in sync with the `poetry.lock` file, they're batch updated periodically, not every time the poetry deps are updated. 

Pragmatic honest answer: when I run into this issue I just run `pip install -U ruff` and things work again :). Note as pure as using a lockfile, but it lets me get things working and get back to work.

**Reply by chloeh13q** [reply]
`pip install -U ruff` solved the problem. Thanks for the fast response!
